sdk/helper/certutil/helpers.go

```diff
@@ -69,21 +69,42 @@ func ParseHexFormatted(in, sep string) []byte {
 	return ret.Bytes()
 }
 
-// GetSubjKeyID returns the subject key ID, e.g. the SHA1 sum
-// of the marshaled public key
+// GetSubjKeyID returns the subject key ID. The computed ID is the SHA-1 hash of
+// the marshaled public key according to
+// https://tools.ietf.org/html/rfc5280#section-4.2.1.2 (1)
 func GetSubjKeyID(privateKey crypto.Signer) ([]byte, error) {
 	if privateKey == nil {
 		return nil, errutil.InternalError{Err: "passed-in private key is nil"}
 	}
+	return getSubjectKeyID(privateKey.Public())
+}
 
-	marshaledKey, err := x509.MarshalPKIXPublicKey(privateKey.Public())
-	if err != nil {
-		return nil, errutil.InternalError{Err: fmt.Sprintf("error marshalling public key: %s", err)}
-	}
-
-	subjKeyID := sha1.Sum(marshaledKey)
+func getSubjectKeyID(pub interface{}) ([]byte, error) {
+	var publicKeyBytes []byte
+	switch pub := pub.(type) {
+	case *rsa.PublicKey:
+		type pkcs1PublicKey struct {
+			N *big.Int
+			E int
+		}
 
-	return subjKeyID[:], nil
+		var err error
+		publicKeyBytes, err = asn1.Marshal(pkcs1PublicKey{
```

Comment:
At least from my playing in the Playground, it [should be OK](https://play.golang.org/p/ZwfM_tKizsi) to replace this with:

```go
publicKeyBytes, err = asn1.Marshal(*pub)
```

since the field ordering is the same and we can avoid needing to duplicate the struct. In [regular Go](https://github.com/golang/go/blob/master/src/crypto/rsa/rsa.go#L41-L45) the struct matches this local struct. In [BoringCrypto fork of Go](https://github.com/golang/go/blob/dev.boringcrypto/src/crypto/rsa/rsa.go#L46-L52), it merely contains an unsafe pointer, which because it is private, shouldn't be accessible to `asn1.Marshal`.

Thoughts? I agree this is more explicit and the less risky approach (in case `rsa.PublicKey` changes public fields in the future), but I'm expecting that to be unlikely. 

Perhaps @sgmiller could weigh in as well.

Reply:
I don't have a strong opinion, but I'd always trade more safety for terseness.

Reply:
Re: release note, I definitely wouldn't call it a bug, since the original implementation isn't in violation of the standard.

Reply:
I agree, I wouldn't classify it as a bug either. Technically, there is nothing wrong with the existing computation.

Reply:
Let's leave as-is then.